### PR TITLE
Reference the k8s provisioner instead of the outdated in-tree provisioner

### LINKED ISSUE #2487: Quickly creating many PVCs results gets stuck after binding only a few PVCs
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
When many PVCs are quickly created (~100) without any delay, the operator stops responding to the requests to create the PVCs after binding only a few of them (the first six in this repro).

The operator log is filled with the following messages as K8s keeps retrying the operations:
```
I0114 16:41:36.821088       8 controller.go:1072] scheduleOperation[provision-cirros/cirros40[caeb899c-180e-11e9-9adf-0287988802f0]]
I0114 16:41:36.821098       8 controller.go:1072] scheduleOperation[provision-cirros/cirros51[cc7ec20a-180e-11e9-8b32-063df0a8d040]]
I0114 16:41:36.821110       8 controller.go:1072] scheduleOperation[provision-cirros/cirros55[cd1051cd-180e-11e9-8b32-063df0a8d040]]
I0114 16:41:36.821118       8 controller.go:1072] scheduleOperation[provision-cirros/cirros89[d2100f8c-180e-11e9-a678-0a5d884f71ec]]
...
```

After some period of time, a timeout message appears in the log and the influx of retries slows down. 
```
I0114 16:42:32.634844       8 leaderelection.go:198] stopped trying to renew lease to provision for pvc cirros/cirros68, task succeeded
I0114 16:42:32.834410       8 leaderelection.go:198] stopped trying to renew lease to provision for pvc cirros/cirros61, task succeeded
...
```

After these timeouts happen, the PVCs all soon complete the requests and all 100 PVCs show as bound.

**Expected behavior:**
The operator should be able to keep up with the PVC requests.

**How to reproduce it (minimal and precise):**

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod):
* Kubernetes version (use `kubectl version`):
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/Rook/master/toolbox.html)):

### PR DESCRIPTION
**Pull request by travisn**
Signed-off-by: travisn <tnielsen@redhat.com>

**Description of your changes:**
The k8s external provisioner had been committed in-tree with rook due to dependency issues. This provisioner is outdated and causing issues such as hanging PVC requests. Now we are referencing the external provisioner code and there is no more need to copy the code in-tree. This change does allow us now to create many PVCs without hitting a hang.

The Rook `go` dependencies now include K8s v12.0 and the external provisioner v1.0. These were the most recent tags I could find to work. The K8s v13.0 tag and the provisioner v2.1 tags did compile. However, at runtime there was an issue with conflicting flags in `glog`. It looks like glog has been removed from all k8s code in favor of the package `k8s.io/klog`. However, the external provisioner has not yet been updated to shed glog. We will need that update before we can get on the latest version.

**Which issue is resolved by this Pull Request:**
Resolves #2487 #2007

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [x] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

### CODE REVIEW

File: cluster/charts/rook-ceph/templates/clusterrole.yaml [hunk 1]
```diff
@@ -66,6 +66,7 @@ rules:
   # PVs and PVCs are managed by the Rook provisioner
   - persistentvolumes
   - persistentvolumeclaims
+  - endpoints
```
Comment (jbw976) [inline cluster/charts/rook-ceph/templates/clusterrole.yaml]: is `endpoints` something that the updated external provisioner code is using?

**Comment by humblec** [comment]
>The K8s v13.0 tag and the provisioner v2.1 tags did compile. However, at runtime there was an issue with conflicting flags in glog. It looks like glog has been removed from all k8s code in favor of the package k8s.io/klog>

@travisn  This is true. 

> It looks like glog has been removed from all k8s code in favor of the package k8s.io/klog. However, the external provisioner has not yet been updated to shed glog. We will need that update before we can get on the latest version.
>
Its happening and PR is already out https://github.com/kubernetes-incubator/external-storage/pull/1106 :)

**Comment by travisn** [comment]
@humblec Is the klog change also in progress for https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner? I see the change has merged in https://github.com/kubernetes-incubator/external-storage/pull/1106, though rook is not referencing that one.

**Comment by travisn** [comment]
@jbw976 i also tested that the failover and fencing is working with these changes. When a pod is failed over to another node, I see messages such as the following in the agent log until the old pod releases the lock:
```
019-02-06 20:18:05.767231 I | flexvolume: volume attachment record rook-ceph-system/pvc-bd2fe8bc-2a4b-11e9-b15a-08002709ffb0 exists for pod: default/wordpress-mysql-6887bf844f-xnf52
2019-02-06 20:18:05.773319 E | flexdriver: Attach volume replicapool/pvc-bd2fe8bc-2a4b-11e9-b15a-08002709ffb0 failed: failed to attach volume pvc-bd2fe8bc-2a4b-11e9-b15a-08002709ffb0 for pod default/wordpress-mysql-6887bf844f-g2hr8. Volume is already attached by pod default/wordpress-mysql-6887bf844f-xnf52. Status Running
```

After the old pod releases the lock, the new pod comes up fine.

**Comment by jbw976** [comment]
@travisn RE: those `Volume is already attached` messages, how long did you see those for? are we talking seconds or minutes? any difference from failover scenario without these changes?

**Comment by travisn** [comment]
@jbw976 No differences that I could see. I tried a couple scenarios where it took many minutes for the old pod to terminate, or where it only took a matter of seconds. Both times, I saw the message `already attached` in the log until the lock was freed, then the new pod started up. I also hit #1507 where we need to determine how to handle fencing when a node is completely lost, but that's a separate issue from this change.

**Pull request merged**